rpcs3/Emu/RSX/GCM.h

```diff
@@ -212,6 +212,169 @@ namespace rsx
 	};
 
 	texture_magnify_filter to_texture_magnify_filter(u8 in);
+
+	enum class stencil_op : u8
+	{
+		keep,
+		zero,
+		replace,
+		incr,
+		decr,
+		invert,
+		incr_wrap,
+		decr_wrap,
+	};
+
+	stencil_op to_stencil_op(u16 in);
+
+	enum class blend_equation : u8
+	{
+		add,
+		min,
+		max,
+		substract,
+		reverse_substract,
+		reverse_substract_signed,
+		add_signed,
+		reverse_add_signed,
+	};
+
+	blend_equation to_blend_equation(u16 in);
+
+	enum class blend_factor : u8
+	{
+		zero,
+		one,
+		src_color,
+		one_minus_src_color,
+		dst_color,
+		one_minus_dst_color,
+		src_alpha,
+		one_minus_src_alpha,
+		dst_alpha,
+		one_minus_dst_alpha,
+		src_alpha_saturate,
+		constant_color,
+		one_minus_constant_color,
+		constant_alpha,
+		one_minus_constant_alpha,
+	};
+
+	blend_factor to_blend_factor(u16 in);
+
+	enum class logic_op : u8
+	{
+		clear,
+		and,
```

Comment:
c++ has many cool (or "cool") hidden stuff like this =) http://en.cppreference.com/w/cpp/language/operator_alternative